### PR TITLE
Force cancel all events on client

### PR DESCRIPTION
**Pull request by yschimke**
Once a test leaves a client running, all other tests will fail.  But leaving things running can impact other tests, so make another attempt to cleanup.

### CODE REVIEW

**Comment by yschimke** [comment]
@swankjesse It's internal class, but you might not want to expose.

File: okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt [hunk 1]
```diff
@@ -242,9 +242,9 @@ class TaskRunner(
     }
   }
 
-  private fun cancelAll() {
+  fun cancelAll() {
     for (i in busyQueues.size - 1 downTo 0) {
-      readyQueues[i].cancelAllAndDecide()
+      busyQueues[i].cancelAllAndDecide()
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt]: Found bug

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/concurrent/TaskRunner.kt]: ooooh

File: okhttp-testing-support/src/main/java/okhttp3/OkHttpClientTestRule.kt [hunk 2]
```diff
@@ -88,15 +89,18 @@ class OkHttpClientTestRule : TestRule {
...
-      assertThat(queue.idleLatch().await(1_000L, TimeUnit.MILLISECONDS))
-          .withFailMessage("Queue still active after 1000 ms")
-          .isTrue()
+    try {
+      for (queue in TaskRunner.INSTANCE.activeQueues()) {
+        assertTrue("Queue still active after 1000 ms",
+            queue.idleLatch().await(1_000L, TimeUnit.MILLISECONDS))
+      }
+    } finally {
+      TaskRunner.INSTANCE.cancelAll()
```
Comment (swankjesse) [inline okhttp-testing-support/src/main/java/okhttp3/OkHttpClientTestRule.kt]: should this be an if/else thing?

```
for (queue in TaskRunner.INSTANCE.activeQueues()) {
  if (!queue.idleLatch().await(1_000L, TimeUnit.MILLISECONDS)) {
    TaskRunner.INSTANCE.cancelAll() 
    fail("Queue still active after 1000 ms")
  }
}
```
that way we don’t cancelAll() when it’s succeeding?

File: okhttp-testing-support/src/main/java/okhttp3/OkHttpClientTestRule.kt [hunk 3]
```diff
@@ -88,15 +89,18 @@ class OkHttpClientTestRule : TestRule {
     testClient?.let {
       val connectionPool = it.connectionPool
       connectionPool.evictAll()
-      assertThat(connectionPool.connectionCount()).isEqualTo(0)
+      assertEquals(0, connectionPool.connectionCount())
```
Comment (yschimke) [inline okhttp-testing-support/src/main/java/okhttp3/OkHttpClientTestRule.kt]: This is due to assertj relying on Duration and failing on Android.

File: android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt [hunk 4]
```diff
@@ -167,6 +167,7 @@ class OkHttpTest {
       }
     } finally {
       Security.removeProvider("Conscrypt")
+      client.close()
```
Comment (yschimke) [inline android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt]: FWIW these tests triggered the overall bug.  They have a client build unchained from OkHttpClientTestRule.

**Pull request merged**